programs/ssl/ssl_client2.c

```diff
@@ -700,41 +742,65 @@ int main( int argc, char *argv[] )
             else if( strcmp( q, "4096" ) == 0 )
                 opt.mfl_code = MBEDTLS_SSL_MAX_FRAG_LEN_4096;
             else
+            {
+                mbedtls_printf( "Invalid value for option %s (must be 512, 1024, 2048 or 4096)\n",
+                                p );
                 goto usage;
+            }
         }
         else if( strcmp( p, "trunc_hmac" ) == 0 )
         {
             switch( atoi( q ) )
             {
                 case 0: opt.trunc_hmac = MBEDTLS_SSL_TRUNC_HMAC_DISABLED; break;
                 case 1: opt.trunc_hmac = MBEDTLS_SSL_TRUNC_HMAC_ENABLED; break;
-                default: goto usage;
+                default:
+                    mbedtls_printf( "Invalid value for option %s (must be 0 or 1)\n",
+                                p );
+                    goto usage;
             }
         }
         else if( strcmp( p, "hs_timeout" ) == 0 )
         {
-            if( ( p = strchr( q, '-' ) ) == NULL )
-                goto usage;
-            *p++ = '\0';
+            char *r = strchr( q, '-' );
+            if( r == NULL )
+                goto hs_timeout_usage;
+            *r++ = '\0';
             opt.hs_to_min = atoi( q );
-            opt.hs_to_max = atoi( p );
+            opt.hs_to_max = atoi( r );
             if( opt.hs_to_min == 0 || opt.hs_to_max < opt.hs_to_min )
+            {
+            hs_timeout_usage:
+                mbedtls_printf( "Invalid value for option %s (must be MIN-MAX with 0<MIN<=MAX)\n",
```

Comment:
Would it be worth printing the values of `opt.hs_to_max` and `opt.hs_to_min` when available?

Reply:
I don't understand why we'd print them.

Reply:
Might it help to print the min and max if min and max were defined properly, but the timeout itself was out of range?